src/api/SubjectsApi.ts

```diff
@@ -77,29 +107,22 @@ function toApiUpdateRequest( subject: Subject ): UpdateBody {
 		id: subject.id,
 	};
 
-	if ( subject.fields.date ) {
-		request = {
-			...request,
-			rawDate: subject.fields.date.rawValue,
-			parsedDate: subject.fields.date.parsedValue.toISOString(),
-		};
-	}
+	Object.entries( subject.fields ).forEach( ( [ fieldName, field ] ) => {
+		const capitalizedFieldName =
+			fieldName.charAt( 0 ).toUpperCase() + fieldName.slice( 1 );
 
-	if ( subject.fields.title ) {
-		request = {
-			...request,
-			rawTitle: subject.fields.title.rawValue,
-			parsedTitle: subject.fields.title.parsedValue,
-		};
-	}
+		let parsedValue = field.parsedValue;
+		// Handle special cases
+		if ( fieldName === 'date' && field.parsedValue instanceof Date ) {
```

Comment:
Instead of using the field's name, I think it would make sense to instead use its type, which we would look for in the schema. Similarly to what we're doing [here](https://github.com/WordPress/try-wordpress/blob/unhardcode_fields/src/api/SubjectsApi.ts#L72).

Reply:
you mean `field.type == FieldType.Date` or more?

Reply:
Sorry, should have made my first comment clearer. What I meant was the following.

Here, we have special logic to handle fields of type `date`, but we're only running the logic when the field is named `date`. I think this logic should run regardless of what the field is called, for example, if the field is called `created_at`, the logic should still run.

This would mean the condition would use the type of the field instead of `fieldName`. So indeed something like `field.type === FieldType.Date`.